server/game/Effects/DynamicEffect.js

```diff
@@ -15,7 +17,7 @@ class DynamicEffect extends StaticEffect {
     recalculate(target) {
         let oldValue = this.getValue(target);
         this.values[target.uuid] = this.calculate(target, this.context);
-        return oldValue !== this.values[target.uuid];
+        return !_.isEqual(oldValue, this.values[target.uuid]);
```

Comment:
I'd prefer to reduce underscore usage rather than increasing it, what is this trying to achieve?

Reply:
This is the first case where the value being used for dynamic recalculation is a dictionary rather than a simple value, and the normal `===` doesn't work for it.  I messed around with trying to figure out whether it's a dictionary and just do a more in-depth comparison in that case but it seems underscore already does that so I didn't want to reinvent.  But let me know if you have other suggestions!

Reply:
I see, thanks. If that's more important than having less code, I could introduce a new kind of effect builder, like `flexibleProperties` or something, that handles this.

Reply:
I think I understand now. Yes please that would be preferable.

Reply:
@cryogen I added `flexibleProperties` (also needed for a new revenant.  Please take another look!

Reply:
LGTM, I'll ask @romanogit so have a look if he has capacity